File: server/datasets/controlled_benchmarks/python/multi_student_cohorts/cohort_b2/organic_5.py
# Two Sum (hash map) - organic submission 5
# Manual while loop with an index counter instead of enumerate.

def two_sum(nums, target):
    seen = {}
    i = 0
    while i < len(nums):
        complement = target - nums[i]
        if complement in seen:
            return [seen[complement], i]
        seen[nums[i]] = i
        i += 1
    return None


if __name__ == "__main__":
    nums = [10, 15, 3, 7]
    target = 17
    print("Indices:", two_sum(nums, target))

File: server/datasets/controlled_benchmarks/java/multiple_files/multiple_b2/organic_5.java
// Two Sum (hash map) - organic submission 5
// Manual while loop with an index counter.

import java.util.Arrays;
import java.util.HashMap;
import java.util.Map;

class Solution {
    public static int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> seen = new HashMap<>();
        int i = 0;
        while (i < nums.length) {
            int complement = target - nums[i];
            if (seen.containsKey(complement)) {
                return new int[]{seen.get(complement), i};
            }
            seen.put(nums[i], i);
            i++;
        }
        return null;
    }

    public static void main(String[] args) {
        int[] nums = {10, 15, 3, 7};
        int target = 17;
        System.out.println("Indices: " + Arrays.toString(twoSum(nums, target)));
    }
}
